### PR TITLE
implement supporter skilled impact tab

### PR DESCRIPTION
**Pull request by shravanhariharan2**
# Description

Add skilled impact tab for supporters.

Includes:
- viewing and adding the supporter's current skills
- all campaigns that match the supporter's skills
- the skilled groups the supporter is part of
- the applications for skilled impact requests that the supporter has submitted

iOS:

IMG_2227

IMG_2228

IMG_2229

Android:

skill_tab_1

skill_tab_2

skill_tab_3

## Checklist

Remove any items which are not applicable.

- [x] I have performed a self-review of my own code.

### CODE REVIEW

File: components/SkilledImpact/elements/SkillGroupContent.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  toggleExpand = () => {
+    this.setState({ expanded: !this.state.expanded });
+  };
+
+  toTitleCase = (str) => {
+    return str
+      .split('_')
+      .map((w) => w[0].toUpperCase() + w.substr(1).toLowerCase())
+      .join(' ');
+  };
```
Comment (dacruz21) [inline components/SkilledImpact/elements/SkillGroupContent.js:81]: Use `constants/Skills.js` to get human-readable skills instead.

File: screens/supporter-skillimpact-screen/SupporterSkillImpactScreen.js [hunk 2]
```diff
@@ -1,38 +1,109 @@
...
+      // Placeholder for skills, remove once Skills Creation screen is made
+      if (!skills) {
+        skills = ['ARCHITECTURE', 'DRONE'];
+      }
+      skills.forEach(async (skill) => {
+        await this.props.getCampaignsBySkill(skill);
+        this.setState({
+          campaigns: this.state.campaigns.concat(this.props.campaignsBySkill),
+        });
+      });
```
Comment (dacruz21) [inline screens/supporter-skillimpact-screen/SupporterSkillImpactScreen.js:56]: You can edit your skills in the edit profile page (right-most tab, gear icon in top right) now, so can we get rid of this?

File: screens/supporter-skillimpact-screen/SupporterSkillImpactScreen.js [hunk 3]
```diff
@@ -1,38 +1,109 @@
...
+    }
   }
 }
 
+const styles = StyleSheet.create({
+  container: {
+    justifyContent: 'flex-start',
+    alignItems: 'center',
+  },
+});
```
Comment (dacruz21) [inline screens/supporter-skillimpact-screen/SupporterSkillImpactScreen.js:97]: Any reason this isn't in `constants/` like the rest of the styles?

File: components/SkilledImpact/elements/ApplicationContent.js [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+  };
+
+  render() {
+    return (
+      <View style={styles.itemContainers}>
+        <TouchableOpacity
+          style={styles.itemTitleRow}
+          onPress={this.toggleMainExpand}
+        >
+          <Sync />
```
Comment (dacruz21) [inline components/SkilledImpact/elements/ApplicationContent.js:43]: This should be a page icon, take a look at the mockup in Monday. Right now it kinda looks like another section of applications, which is a little awkward.

File: components/SkilledImpact/elements/ApplicationContent.js [hunk 5]
```diff
@@ -18,6 +18,15 @@ class ApplicationContent extends React.Component {
...
     };
+    this.noDecisionSubmissions = props.submissions.map(
+      (submission) => submission.submission.decision === 'PENDING'
+    );
+    this.selectedSubmissions = props.submissions.map(
+      (submission) => submission.submission.decision === 'ACCEPTED'
+    );
+    this.notSelectedSubmissions = props.submissions.map(
+      (submission) => submission.submission.decision === 'DENIED'
+    );
```
Comment (dacruz21) [inline components/SkilledImpact/elements/ApplicationContent.js:29]: These should be using `.filter()`, not `.map()`, right?

Comment (shravanhariharan2) [reply components/SkilledImpact/elements/ApplicationContent.js:29]: oh yeah whoops, thanks for the catch

**Pull request merged**